### ISSUE TITLE
Acces to USPIO-6 data

### ISSUE THREAD

**Issue by ebannier2018** [fli-iam/shanoir-ng#725]
Hi, 
Below a few problems : 
1. It seems the USPIO-6 data is not accessible anymore via the Medical Data => Studies menu, 
2. I managed to access one dataset via SOLR by checking the study so I know the data is there but this is unstable : 
I keep getting the Unexpected error occured when using the text field, not explicit to allow me to understand what I am doing wrong in the request (apparently "-")
3. I finally do not get any error message with the studyName:USPIO*; but also no results 

In summary : 
Can you check what is wrong with the access to the USPIO-6 Data via he Medical Data => Studies menu ? 
Can you make the error message more explicit ? 
Can you check what is wrong with the SOLR search ? 
Let me know if something is unclear, 
Thanks in advance, 
Elise

**Comment by ebannier** [comment]
Just an update, when I mean not accessible : the subjects are there, the examination dates also but not the list of sequences.

**Comment by yyao1231** [comment]
Hi Elise,

Thanks for your feedback. 
For the 2nd and the 3rd question about Solr search, the special characters like "-" is not yet supported in this version of search bar. So, to get the data, try with "studyName:USPIO*;" syntaxe, and below is what I got: 
![image](https://user-images.githubusercontent.com/12045953/102757092-12296400-4371-11eb-9bb9-88720c1a43e4.png)

The dicom/nii download works fine for me via the solr search UI. 

For the first question, I confirmed that the study>subject>examination tree is broken. I got errors when I tried to get data by this request for example: https://shanoir.irisa.fr/shanoir-ng/datasets/examinations/subject/12/study/2 
```
code: 500
details: null
message: "could not deserialize"
```
For this bug, I will check what is recently committed to get the tree broken. It seems that the data could be accessed by the BIDS tree.

I will let you know about the study>subject>examination tree bug. Anyway, the data could be accessed by the BIDS tree and the solr search menu. 

Yao

**Comment by ebannier** [comment]
Hi,

Thanks Yao. Yes the SOLR download also worked fine for me when I removed the "-" but the error message was not explicit and I could solve this with a hint from Isabelle. Could you add some comment next to the SOLR search windows to avoid this problem in the future for other users? 
As for 1. It remains sometimes clearer for us to access data via the Medical data ==> Studies tree and it would be nice to have the USPIO data back this way. As discussed, SOLR search does not completely answer our needs in its current implementation. 

Elise

**Comment by yyao1231** [comment]
> Hi,
> 
> Thanks Yao. Yes the SOLR download also worked fine for me when I removed the "-" but the error message was not explicit and I could solve this with a hint from Isabelle. Could you add some comment next to the SOLR search windows to avoid this problem in the future for other users?

No pb. I just added it in the tooltip at the right of the "Solr Search" title. You will see them at the next deployment.
> As for 1. It remains sometimes clearer for us to access data via the Medical data ==> Studies tree and it would be nice to have the USPIO data back this way. As discussed, SOLR search does not completely answer our needs in its current implementation.
> 
Yes I understand. The pb occurs only in the tree of "USPIO-6" study. I will try to reproduce it on my local dev environnement, if I could somehow get the same data. I will let you know.
> Elise

**Comment by yyao1231** [comment]
Elise,

This bug is very interesting! I got some dicom files via solr search for the studyId=2 (USPIO-6) and subjectId=12 (01001); I generated dicomdir for these dicom files and then imported them to my local dev environnement. But the bug could not be reproduced. Everything is displayed normally. 
![image](https://user-images.githubusercontent.com/12045953/102806594-7540e800-43bd-11eb-915d-066f8ffbde7c.png)

Then I noticed that on Neurinfo, for the USPIO-6 study, only the patients 01*** have a pb with study tree display, but not the other patients below.
![image](https://user-images.githubusercontent.com/12045953/102807384-d4532c80-43be-11eb-817d-18560e75e74c.png)
Maybe you could help me to know the differences between these patients? Especially how the examinations are named? As the display pb seems to happen on their examination date- comment -id pattern (500 could not deserialize). 
I'm connected to the neurinfo database via ssh + vpn, and I got all datasets with studyId=2 and subjectId=12. I could check one by one if there is something wrong with the data, but that could take time..
Otherwise, do you remember when you have encountered this bug for the first time? Only after the shanoir-ng's first deployment?

Thanks,
Yao

**Comment by ebannier** [comment]
Hi, 

The first 2 digits code the center so all data from center 01 is from one center, namely Rennes for patients. 06 is Rennes for healthy controls and we can access the list of sequences via the Studies tree. Data from subject 02001 is also not accessible though. Could it be that something went wrong during the migration and failed for the first sets of subjects (01...to 02001 included?)   

I had not noticed this bug on shanoir-old, the data was definitely accessible. I only noticed it on Shanoir NG recently, 

Elise

btw, when the issue is solved, we should remove the screenshots.

**Comment by yyao1231** [comment]
Logs on server:
```
Caused by: java.io.StreamCorruptedException: invalid stream header: 32303132
       at org.shanoir.ng.datasetacquisition.dto.mapper.ExaminationDatasetAcquisitionMapperImpl_.datasetProcessingListToDatasetProcessingDTOList(ExaminationDatasetAcquisitionMapperImpl_.java:134)
        at org.shanoir.ng.datasetacquisition.dto.mapper.ExaminationDatasetAcquisitionMapperImpl_.datasetToDatasetAndProcessingsDTO(ExaminationDatasetAcquisitionMapperImpl_.java:158)
        at org.shanoir.ng.datasetacquisition.dto.mapper.ExaminationDatasetAcquisitionMapperImpl_.datasetListToDatasetAndProcessingsDTOList(ExaminationDatasetAcquisitionMapperImpl_.java:170)
        at org.shanoir.ng.datasetacquisition.dto.mapper.ExaminationDatasetAcquisitionMapperImpl_.datasetAcquisitionToExaminationDatasetAcquisitionDTO(ExaminationDatasetAcquisitionMapperImpl_.java:70)
        at org.shanoir.ng.datasetacquisition.dto.mapper.ExaminationDatasetAcquisitionDecorator.datasetAcquisitionToExaminationDatasetAcquisitionDTO(ExaminationDatasetAcquisitionDecorator.java:54)
        at org.shanoir.ng.datasetacquisition.dto.mapper.ExaminationDatasetAcquisitionDecorator.datasetAcquisitionsToExaminationDatasetAcquisitionDTOs(ExaminationDatasetAcquisitionDecorator.java:45)
        at org.shanoir.ng.examination.dto.mapper.ExaminationMapperImpl_.examinationToSubjectExaminationDTO(ExaminationMapperImpl_.java:138)
        at org.shanoir.ng.examination.dto.mapper.ExaminationMapperImpl_.examinationsToSubjectExaminationDTOs(ExaminationMapperImpl_.java:65)
        at org.shanoir.ng.examination.dto.mapper.ExaminationMapperImpl.examinationsToSubjectExaminationDTOs(ExaminationMapperImpl.java:33)
        at org.shanoir.ng.examination.controler.ExaminationApiController.findExaminationsBySubjectIdStudyId(ExaminationApiController.java:153)

```

**Comment by yyao1231** [comment]
I found something wrong about the data migration. For example, for the datasets id=37445, 37446, 37447, 37448, 37449 and 37455, they don't have comment column on db. I got the same error when I try to access to their dataset details page. But one strange thing: for Dataset class in shanoir-ng, we don't have the "comment" column anymore. I'm pretty sure that it exists in shanoir old and in db on prod. So I have to discuss with everyone about that since I was not here.. Did we get some error during the data migration?

**Comment by Nunien** [comment]
Hi Yao, 

Dataset data model has been modified between old and ng. 

comment is now in DatasetMetadata. 

note that Dataset has two DatasetMetadata : 
* originMetadata 
* updatedMetadata 

see you, 

Julien

**Comment by yyao1231** [comment]
Thanks Julien! 
Anybody knows if the data migration script is executed without any error? 
Maybe I've missed some changes.. But when I connect to the db on prod, the DATASET table looks like the shanoir old's way. There is no DatasetMetadata table on shanoirdb on prod. 
By the way, I used " dk-mysql prod shanoirngdb" command to access to the shanoirdb on prod. 
 `prod         datasets import keycloak pacsdb preclinical shanoirdb studies`

**Comment by Nunien** [comment]
My guess is that you are using the old db. There is a dk-mysql command where you specify the microservice you want to use, this is how you connect to ng databases. I'm sorry I don't have it right now I'm still waiting for my laptop to be repaired.

**Comment by yyao1231** [comment]
Thanks Julien I got the right command for ng db now.

**Comment by yyao1231** [comment]
There is actually a bug with dataset processing deserialization, caused by processing_date. see #685

**Comment by Nunien** [comment]
@yyao1231 thanks Yao I was looking for this!

**Comment by yyao1231** [comment]
corrected on prod

**Comment by Nunien** [comment]
@ebannier , do you confirm that it works fine now?

**Comment by ebannier** [comment]
I tested the access to one dataset, subject 01007, 2D FLASH, nifti download and it worked fine, yes. No other exhaustive testing though.
